src/api/ammPools.ts

```diff
@@ -0,0 +1,31 @@
+import { PoolId } from '@ergolabs/ergo-dex-sdk';
+import { map, Observable, publishReplay, refCount, switchMap } from 'rxjs';
+
+import { AmmPool } from '../common/models/AmmPool';
+import { selectedNetwork$ } from '../network/network';
+
+export const ammPools$ = selectedNetwork$.pipe(
+  switchMap((network) => network.ammPools$),
+  publishReplay(1),
+  refCount(),
+);
+
+export const getAmmPoolById = (
+  ammPoolId: PoolId,
+): Observable<AmmPool | undefined> =>
+  ammPools$.pipe(
+    map((pools) => pools.find((position) => position.id === ammPoolId)),
+  );
+
+const byPair = (xId: string, yId: string) => (p: AmmPool) =>
+  (p.x.asset.id === xId || p.y.asset.id === xId) &&
+  (p.x.asset.id === yId || p.y.asset.id === yId);
+export const getAmmPoolsByPair = (
+  xId: string,
+  yId: string,
+): Observable<AmmPool[]> =>
+  ammPools$.pipe(
+    map((pools) => pools.filter(byPair(xId, yId))),
+    publishReplay(1),
+    refCount(),
```

Comment:
What if we start to avoid deprecated `rx` functions?
Why are they deprecated?

Reply:
new rxjs lib has new connectable observable api. I think we rewrite all shared observables soon, but i need read new api and think about structures one more time.